src/uu/hashsum/src/hashsum.rs

```diff
@@ -65,7 +65,32 @@ fn detect_algo(
         "sha256sum" => ("SHA256", Box::new(Sha256::new()) as Box<dyn Digest>, 256),
         "sha384sum" => ("SHA384", Box::new(Sha384::new()) as Box<dyn Digest>, 384),
         "sha512sum" => ("SHA512", Box::new(Sha512::new()) as Box<dyn Digest>, 512),
-        "b2sum" => ("BLAKE2", Box::new(Blake2b::new()) as Box<dyn Digest>, 512),
+        "b2sum" => match matches.get_one::<usize>("length") {
+            Some(length_in_bits) => {
+                if *length_in_bits > 512 {
+                    crash!(1, "Invalid length (maximum digest length is 512 bits)")
+                }
+
+                // --length=0 falls back to default behavior
+                if *length_in_bits == 0 {
+                    return ("BLAKE2", Box::new(Blake2b::new()) as Box<dyn Digest>, 512);
+                }
```

Comment:
I would move this functionality to its own arm `Some(0)`.

Reply:
I combined it with `None` arm in commit fdbebfa